docs/src/user/tasks.rst

```diff
@@ -18,6 +18,111 @@ Predictions are generated by *Model Developers* when they define :meth:`rafiki.m
 and received by *Application Users* as predictions to their queries sent to *Inference Jobs*.
 
 
+QUESTION_ANSWERING
+--------------------------------------------------------------------
+
+
+Dataset Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`dataset-type:QUESTION_ANSWERING`
+
+
+Dataset can be used to finetune the SQuAD pre-trained Bert model. 
+
+- The dataset zips folders containing JSON files. JSON files under different level folders will be automaticly read all together. 
+
+An example of the dataset structure:
+
+.. code-block:: text
+
+    /DATASET_NAME.zip
+    │
+    ├──FOLDER_NAME_1                                              # first level folder
+    │  └──FOLDER_NAME_2                                           # second level folder, not necessarily to be included
+    │      └──FOLDER_NAME_3                                       # third level folder, not necessarily to be included
+    │           ├── 003d2e515e1aaf06f0052769953e8.json            # JSON file name is a random combination of either alphabets/numbers or both
+    │           ├── 00a407540a8bdd.json
+    │           ...
+    │
+    ├──FOLDER_NAME_4                                              # first level folder
+    │  ├── 0015023cc06b5362d332b3.json
+    │  ├── 001b4a31684c8fc6e2cfbb70304354978317c429.json
+    │  ...
+    ...
+    │
+    └──metadata.csv                                          # if additional information is provided for above JSON files, user can add a metadata.csv
+
+- JSON file includes ``body_text``, providing list of paragraphs in full body which can be used for question answering. ``body_text`` can contain different entries, only the "text" field of each entry will be read.
+
+The essential structure of JSON file is:
+
+.. code-block:: text
+
+    {
+        "body_text": [                      # list of paragraphs in full body
+            {                               
+                "text": <str>,              # text body for first entry, this is must-have 
```

Comment:
I propose that the 'text' field contains ethier paragraph extracted from papers, or both question and information paragraph, since the QA focusing on various purpose, so we better make it general.

Reasons are as follows:
1. take our covid19 dataset into consideration, the multiple 'text' fields are paragraphs extracted, each JSON per paper, and each paragraph per `text`, where there is no `question` `context` nor `answer` fields, just paragraphs as corpus.
2. but available QA datasets like SQuAD2.0, MediQA2019, Medical Question Answer Data, contain `context` and `answer` for each `question`. In this convention, we'd better not make `text` too particular for these datasets.
3. MedQuAD provides a bunch of question-answer pairs, without `context` field.  I think we can take the question-answer pairs as information paragraphs in the most general structure, that is, each pair be feed to each `text` field in one or multiple JSON, as corpus.

Examples are given in updated Pull Request.

Reply:
the format is too general that it can be applied for any text processing task.
since the task is question answering, there must be a field for the question/answer/context. for some datasets, the context field is just empty.
suppose there are 3 types of formats used by public datasets, A, B, and C.
now if we propose a new format D. we have to convert A->D, B->D and C->D.
if we know that A is more popular, then it is better to just convert B->A and C->A.

if MedQuAD and SQuAD, MediQA2019 share the same format, and convid-19 can be converted into that format, why not use it?

Reply:
MedQuAD does not share the format with other datasets. Neither does the Covid19. 

We can see if all other datasets are able to be converted into the format of SQuAD and MediQA2019. 

So do I start to convert them?